extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourceRuntimeConfig.java

```diff
@@ -124,4 +124,18 @@
      */
     @ConfigItem
     public Optional<String> validationQuerySql;
+
+    /**
+     * The socket factory driver that should be appended as a datasource url property
+     */
+    @ConfigItem
+    public Optional<String> socketFactory;
+
+    /**
+     * General JDBC properties that can configured for a specific JDBC connection
+     *
+     * Should be comma separated, example: <property_name1>=<property_value1>,<property_name2>=<property_value2>
+     */
+    @ConfigItem
+    public Optional<String> additionalJdbcProperties;
```

Comment:
It should probably be a Map<String, String>.

@Sanne WDYT about adding these 2 properties?

Reply:
@gsmet @Sanne I have test this code and I have noticed that the JDBC properties are passed to AgroalDatasource but they are not used anywhere afterwards. Am I missing something here or this is a bug?

Reply:
As I did in my fork, the only way to get through this is to use the jdbcURL property. I am working on a dedicated extension that will enable both Postgres and MySQL db.

See https://github.com/quarkusio/quarkus/pull/5901

Reply:
I understand that your way works @jtama-op but that doesn't make it right. we have to find out what is actually going wrong and fix it, not work around it

Reply:
@gsmet : Is there a way we could speed this up a little bit? It is becoming a bit of a blocker for me

Reply:
@Mihai-B you can still connect to gcp cloud sql datasource using agroal extension with the following properties : 
``quarkus.datasource.url = jdbc:postgresql:///postgres?socketFactory=com.google.cloud.sql.postgres.SocketFactory&cloudSqlInstance=<YOUR_INSTANCE>``

Just be carefull to you service account and authentication.

Reply:
@jtama-op did you run it in native mode?
I get this error in native mode: 
`
Caused by: org.postgresql.util.PSQLException: The SocketFactory class provided com.google.cloud.sql.postgres.SocketFactory could not be instantiated.
at org.postgresql.core.SocketFactoryFactory.getSocketFactory(SocketFactoryFactory.java:43)

Caused by: java.lang.ClassNotFoundException: com.google.cloud.sql.postgres.SocketFactory
at om.oracle.svm.core.hub.ClassForNameSupport.forName(ClassForNameSupport.java:60)`

Reply:
Nope.  Only with JVM.

Reply:
Ok, so ths solution is not complete then...